### PR TITLE
refactor: aes and rsa keys initialization

### PR DESCRIPTION
**Pull request by jeqo**
Changes:
- Moves RSA Path into InputStream mapping into one place
- Creates an AES Encryption Provider out of an RSA Encryption provider.

### CODE REVIEW

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java [hunk 1]
```diff
@@ -56,8 +58,8 @@ private RsaEncryptionProvider(final KeyPair rsaKeyPair) {
         this.rsaKeyPair = rsaKeyPair;
     }
 
-    public static RsaEncryptionProvider of(final InputStream rsaPublicKey,
-                                           final InputStream rsaPrivateKey) {
+    public static RsaEncryptionProvider of(final Path rsaPublicKey,
+                                           final Path rsaPrivateKey) {
```
Comment (ivanyu) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java:62]: Considering https://github.com/aiven/tiered-storage-for-apache-kafka/issues/198, maybe we should keep `InputStream`?

Comment (jeqo) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java:62]: Good point. What about keeping Paths here, and leave constructor with InputStream instead? Then we can add another static `of` for getting PEM content directly from config, wdyt?

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java:62]: Yeah, makes sense

Comment (jeqo) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java]: Wasn't actually the constructor, but I kept the InputStream method on RsaKeysReader added a new one for Paths. LMK how it looks now

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java [hunk 2]
```diff
@@ -91,6 +92,14 @@ public byte[] decryptDataKey(final byte[] bytes) {
 
     static class RsaKeysReader {
 
+        static KeyPair readRsaKeyPair(final Path publicKeyPath, final Path privateKeyPath) {
+            try {
+                return readRsaKeyPair(Files.newInputStream(publicKeyPath), Files.newInputStream(privateKeyPath));
```
Comment (ivanyu) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java:97]: These `InputStream`s must be closed, i.e. need to be in `try`

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java [hunk 3]
```diff
@@ -91,6 +92,14 @@ public byte[] decryptDataKey(final byte[] bytes) {
...
+            } catch (IOException e) {
```
Comment (ivanyu) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/RsaEncryptionProvider.java:98]:
```suggestion
            } catch (final IOException e) {
```

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java [hunk 4]
```diff
@@ -33,7 +36,15 @@ public class AesEncryptionProvider implements Encryption, Decryption {
 
     private final KeyGenerator aesKeyGenerator;
 
-    public AesEncryptionProvider(final KeyGenerator aesKeyGenerator) {
+    public static AesEncryptionProvider of(final RsaEncryptionProvider rsaEncryptionProvider) {
+        try {
+            return new AesEncryptionProvider(rsaEncryptionProvider.keyGenerator());
+        } catch (NoSuchAlgorithmException | NoSuchProviderException e) {
```
Comment (ivanyu) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:42]:
```suggestion
        } catch (final NoSuchAlgorithmException | NoSuchProviderException e) {
```

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java [hunk 5]
```diff
@@ -33,7 +36,15 @@ public class AesEncryptionProvider implements Encryption, Decryption {
 
     private final KeyGenerator aesKeyGenerator;
 
-    public AesEncryptionProvider(final KeyGenerator aesKeyGenerator) {
+    public static AesEncryptionProvider of(final RsaEncryptionProvider rsaEncryptionProvider) {
```
Comment (ivanyu) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:39]: I'm not sure the whole key generator code should at all be in `RsaEncryptionProvider` as it's not used there. This means, `AesEncryptionProvider` and `RsaEncryptionProvider` could be completely independent. The `KEY_SIZE` constant should also be in `AesEncryptionProvider`. Maybe I'm missing something that justifies the current coupling between the two classes?

**Review by ivanyu — CHANGES_REQUESTED** [review]

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java (hunk 5 above)
Comment (jeqo) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:39]: Good point.
Even though we may want to keep these 2 decoupled, most users of these APIs have to initialize them together. I'm touching on that in #191 
Let me know if you feel we should decouple these here and discuss the coupling issues on #191

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:39]: Sure, let's postpone it until #191

Comment (jeqo) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:39]: Oh, I just realized, RSA is not required for AES at all! Let me refactor it based on this suggestions and remove the coupling then

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:39]: Yeah, they meet each other only when we encrypt AES keys with RSA.

**Review by ivanyu — APPROVED** [review]

**Pull request merged**